drivers/mrf24j40/mrf24j40_netdev.c

```diff
@@ -0,0 +1,578 @@
+            mrf24j40_set_option(dev, MRF24J40_OPT_TELL_TX_START,
+                                ((bool *)val)[0]);
+            res = sizeof(netopt_enable_t);
+            break;
+
+        case NETOPT_TX_END_IRQ:
+            mrf24j40_set_option(dev, MRF24J40_OPT_TELL_TX_END,
+                                ((bool *)val)[0]);
+            res = sizeof(netopt_enable_t);
+            break;
+
+        case NETOPT_CSMA:
+            mrf24j40_set_option(dev, MRF24J40_OPT_CSMA,
+                                ((bool *)val)[0]);
+            res = sizeof(netopt_enable_t);
+            break;
+
+        case NETOPT_CSMA_RETRIES:
+            if ((len > sizeof(uint8_t)) ||
+                (*((uint8_t *)val) > 5)) {
+                res = -EOVERFLOW;
+            }
+            else if (dev->netdev.flags & MRF24J40_OPT_CSMA) {
+                /* only set if CSMA is enabled */
+                mrf24j40_set_csma_max_retries(dev, *((uint8_t *)val));
+                res = sizeof(uint8_t);
+            }
+            break;
+
+        case NETOPT_CCA_THRESHOLD:
+            if (len > sizeof(int8_t)) {
+                res = -EOVERFLOW;
+            }
+            else {
+                mrf24j40_set_cca_threshold(dev, *((int8_t *)val));
+                res = sizeof(int8_t);
+            }
+            break;
+
+        default:
+            break;
+    }
+    /* try netdev2 building flags */
+    if (res == -ENOTSUP) {
+        res = netdev2_ieee802154_set((netdev2_ieee802154_t *)netdev, opt,
+                                     val, len);
+    }
+    return res;
+}
+
+static void _isr(netdev2_t *netdev)
+{
+    mrf24j40_t *dev = (mrf24j40_t *) netdev;
+
+#ifdef MODULE_NETSTATS_L2
+#endif
+
+    /* update pending bits */
+    mrf24j40_update_tasks(dev);
+    DEBUG("[mrf24j40] INTERRUPT (pending: %x),\n", dev->pending);
+    /* TRANSMIT INTERRUPT OCCURED */
+    if (dev->pending & MRF24J40_TASK_TX_READY) {
+        dev->pending &= ~(MRF24J40_TASK_TX_READY);
+        DEBUG("[mrf24j40] EVT - TX_END\n");
+#ifdef MODULE_NETSTATS_L2
+        if (netdev->event_callback && (dev->netdev.flags & MRF24J40_OPT_TELL_TX_END)) {
+            uint8_t txstat = mrf24j40_reg_read_short(dev, MRF24J40_REG_TXSTAT);
+            /* transmision failed */
+            if (txstat & MRF24J40_TXSTAT_TXNSTAT) {
+                /* TX_NOACK - CCAFAIL */
+                if (txstat & MRF24J40_TXSTAT_CCAFAIL) {
+                    netdev->event_callback(netdev, NETDEV2_EVENT_TX_MEDIUM_BUSY);
+                    DEBUG("[mrf24j40] TX_CHANNEL_ACCESS_FAILURE\n");
+                }
+                /* check max retries */
+                else if (txstat & 0xc0) {
+                    netdev->event_callback(netdev, NETDEV2_EVENT_TX_NOACK);
+                    DEBUG("[mrf24j40] TX NO_ACK\n");
+                }
+            }
+            else {
+                netdev->event_callback(netdev, NETDEV2_EVENT_TX_COMPLETE);
+            }
+        }
+#endif
```

Comment:
I assumed it was irrelevant since in `sys/net/gnrc/link_layer/netdev2/gnrc_netdev2.c` the switch statement also has the same ifdef around the handlers of those callback options (sys/net/gnrc/link_layer/netdev2/gnrc_netdev2.c 76:83. If you want the ifdef removed or minimized, I'm okay with that, but as it is now, it saves an SPI transfer when L2 stats are not enabled and I would like to keep that